doc/source/development/rfc/rfc86_column_oriented_api.rst

```diff
@@ -0,0 +1,692 @@
+or using ResetReading()/GetNextFeature() while using a ArrowArrayStream is
+strongly discouraged and may lead to unexpected results. As a rule of thumb,
+no OGRLayer methods that affect the state of a layer should be called on a
+layer, while an ArrowArrayStream on it is active.
+
+A potential usage can be:
+
+.. code-block:: cpp
+
+    struct ArrowArrayStream stream;
+    if( !poLayer->GetArrowStream(&stream, nullptr))
+    {
+        fprintf(stderr, "GetArrowStream() failed\n");
+        exit(1);
+    }
+    struct ArrowSchema schema;
+    if( stream.get_schema(&stream, &schema) == 0 )
+    {
+        // Do something useful
+        schema.release(schema);
+    }
+    while( true )
+    {
+        struct ArrowArray array;
+        // Look for an error (get_next() returning a non-zero code), or
+        // end of iteration (array.release == nullptr)
+        //
+        if( stream.get_next(&stream, &array) != 0 ||
+            array.release == nullptr )
+        {
+            break;
+        }
+        // Do something useful
+        array.release(&array);
+    }
+    stream.release(&stream);
+
+The papszOptions that may be provided is a NULL terminated list of key=value
+strings, that may be driver specific.
+
+OGRLayer has a base implementation of GetArrowStream() that is such:
+
+- The get_schema() callback returns a schema whose top-level object returned is
+  of type Struct, and whose children consist in the FID column, all OGR attribute
+  fields and geometry fields to Arrow fields.
+  The FID column may be omitted by providing the INCLUDE_FID=NO option.
+
+  When get_schema() returns 0, and the schema is no longer needed, it must
+  be released with the following procedure, to take into account that it might
+  have been released by other code, as documented in the Arrow C data
+  interface:
+
+  .. code-block:: c
+
+          if( out_schema->release )
+              out_schema->release(out_schema)
+
+
+- The get_next() callback retrieve the next record batch over the layer.
+
+  out_array is a pointer to a ArrowArray structure, that can be in a uninitialized
+  state (the method will ignore any initial content).
+
+  The default implementation uses GetNextFeature() internally to retrieve batches
+  of up to 65,536 features (configurable with the MAX_FEATURES_IN_BATCH=num option).
+  The starting address of buffers allocated by the
+  default implementation is aligned on 64-byte boundaries.
+
+  The default implementation outputs geometries as WKB in a binary field,
+  whose corresponding entry in the schema is marked with the metadata item
+  ``ARROW:extension:name`` set to ``WKB``. Specialized implementations may output
+  by default other formats (particularly the Arrow driver that can return geometries
+  encoded according to the GeoArrow specification (using a list of coordinates).
+  The GEOMETRY_ENCODING=WKB option can be passed to force the use of WKB (through
```

Comment:
Could it in theory (later) also be possible to enable that you can specify `GEOMETRY_ENCODING=GEOARROW` to force the use of this encoding, even if the file itself is not using it?

Reply:
yes one could imagine such an extension